File: ruby/organizing_policy/lib/organizing/invitation_repository.rb
module Organizing
  module InvitationRepository
    def save(invitation) = raise NotImplementedError
    def find_by_id(invitation_id) = raise NotImplementedError
    def find_by_inviting_user_id(user_id) = raise NotImplementedError
    def find_by_invited_user_id(user_id) = raise NotImplementedError
  end
end

File: java/organizing-policy/src/main/java/com/hexagonalchess/organizing/InvitationRepository.java
package com.hexagonalchess.organizing;

import java.util.List;
import java.util.Optional;

public interface InvitationRepository {
    void save(GameInvitation invitation);
    Optional<GameInvitation> findById(InvitationId id);
    List<GameInvitation> findByInvitingUserId(String userId);
    List<GameInvitation> findByInvitedUserId(String userId);
}
